util/src/executor.rs

```diff
@@ -0,0 +1,289 @@
+
+
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+extern crate eventual;
+pub use self::eventual::*;
+
+use std::cmp::{Ord, Ordering};
+use std::collections::binary_heap::BinaryHeap;
+use std::marker::PhantomData;
+use std::sync::{atomic, Arc, Condvar, Mutex};
+use std::thread::{JoinHandle, spawn};
+
+#[macro_export]
+macro_rules! ordering_for {
+	($type_: ident using $field_: ident) => {
+		impl Ord for $type_ {
+			fn cmp(&self, other: &$type_) -> Ordering {
+				self.$field_.cmp(&other.$field_)
+			}
+		}
+		impl PartialOrd for $type_ {
+			fn partial_cmp(&self, other: &$type_) -> Option<Ordering> {
+				self.$field_.partial_cmp(&other.$field_)
+			}
+		}
+		impl Eq for $type_ {}
+		impl PartialEq for $type_ {
+			fn eq(&self, other: &$type_) -> bool {
+				self.$field_.eq(&other.$field_)
+			}
+		}
+	}
+}
+
+/// Some work to be done in a thread pool
+pub trait Task: Ord + Send + 'static {
+	/// Task result
+	type TResult: Send + 'static;
```

Comment:
avoid this convention for naming. gav's rules of naming number 1: make names pronouncable.